commitizen/config/base_config.py

```diff
@@ -0,0 +1,34 @@
+from typing import Optional
+
+from commitizen.defaults import DEFAULT_SETTINGS
+
+
+class BaseConfig:
+    def __init__(self):
+        self._settings: dict = DEFAULT_SETTINGS.copy()
+        self._path: Optional[str] = None
+
+    @property
```

Comment:
If possible add the typehints of both properties. Thanks

Reply:
Done.